### ISSUE TITLE
Support typedoc

### ISSUE THREAD

**Issue by WimYedema** [erikrose/sphinx-js#40]
I live sphinx for documentation and would like to use it for my typescript project. Could you use typedoc's json output in this extension?

**Comment by erikrose** [comment]
Quite possibly, with a little work. Can typedoc, like jsdoc, emit its findings as JSON? If so, it should be a fairly straightforward matter to put an abstraction layer between the output and the HTML builders in renderers.py. See jsdoc.py for the part that translates jsdoc's JSON output into sphinx-js's internal data structures. You'd have to write something like that as well. I'd be happy to advise someone as they work on a patch.

**Comment by WimYedema** [comment]
Yes, typedoc can emit JSON. Can you give me some pointers on how to use a development version of sphinx-js?

**Comment by erikrose** [comment]
Sure. I assume you're already using virtualenv, which is always a good idea. Where you'd ordinarily say `pip install sphinx-js`, instead use git to check it out, `cd` into that newly checked out directory, and say `pip install -e .`. (If you already have sphinx-js installed, uninstall it before doing any of this.) `-e` is for "editable". Now any change you make in the checkout will take immediate effect. Let me know if you have more questions!

**Comment by WimYedema** [comment]
Hmm... unfortunately the JSON schema generated by typedoc is not quite the same as that for JSDoc (and the typedoc JSON schema is not documented). I should be able to convert one into the other, but it becomes a bit more difficult...

**Comment by erikrose** [comment]
Neither of them is particularly documented, so take heart: I found it possible. :-) There really aren't that many cases, and I believe the tests cover most if not all (though, of course, you'd have to duplicate the tests and add TS annotations (or something similar)).

**Comment by WimYedema** [comment]
I have a working typescript solution. Need to create some tests still though.
I didn't realize that the sphinx JS domain is rather lacking with respect to what JSdoc (and typedoc) support.   I do want to see information like inheritance and interface usage, though. For now I just add them in the "description" part (for typedoc), but I'd rather have a sphinx directive that has fields for these things...

**Comment by erikrose** [comment]
I agree. For now, I'd say shove them in wherever you can make them look best. Ultimately, we should either write our own directive and forget about Sphinx's native stuff or, better still, add more stuff to Sphinx itself.

**Comment by WimYedema** [comment]
I'm having some trouble running nosetests manually. I get the following error:

```
$ nosetests -vv
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
Failure: ModuleNotFoundError (No module named 'parsimonious') ... ERROR
```

Do you know what I'm doing wrong?

**Comment by erikrose** [comment]
If you were running `python setup.py test` before, that installs dependencies in the current working directory. You'll need to install the dependencies in your active Python environment (doing `pip install -e .` is my usual) before running nosetests manually.

**Comment by WimYedema** [comment]
Thanks, that works:) Just pushed my first typedoc test. I mean to split it in 2 parts, (1) typedoc to jsdoc conversion, and to (2) running sphinx_js with typedoc.

On a side-note: A nice way to gitignore my virtualenv is to add a .gitignore with '*' in my virtualenv directory.

**Comment by erikrose** [comment]
Great! Just let me know when you'd like me to review something.

**Comment by WimYedema** [comment]
Finally I managed to get round to it again...
Can you have a look at https://github.com/WimYedema/sphinx-js/tree/typescript-support?
Thanks.

**Comment by erikrose** [comment]
Sure. If you open a pull request, it'll be easier, since it will provide a convenient way to tie together any comments I make.

**Comment by erikrose** [comment]
See PR #62.

**Comment by erikrose** [comment]
Merged #62!
